src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageAuth.cpp

```diff
@@ -58,28 +82,64 @@ void validateContainerName(const String & container_name)
 
 AzureBlobStorageEndpoint processAzureBlobStorageEndpoint(const Poco::Util::AbstractConfiguration & config, const String & config_prefix)
 {
-    std::string storage_url;
-    if (config.has(config_prefix + ".storage_account_url"))
+    String storage_url;
+    String account_name;
+    String container_name;
+    String prefix;
+    if (config.has(config_prefix + ".endpoint"))
+    {
+        String endpoint = config.getString(config_prefix + ".endpoint");
+
+        /// For authenitication methods other than managed identity (Eg: SAS, Workload Identity),
+        /// account name is not present in the endpoint
+        /// 'endpoint_contains_account_name' bool is used to understand how to split the endpoint (default : true)
+
+        bool endpoint_contains_account_name = config.getBool(config_prefix + ".endpoint_contains_account_name", true);
+
+        size_t pos = endpoint.find("//");
+        assert (pos != std::string::npos);
+
+        if (endpoint_contains_account_name)
+        {
+            size_t pos_begin = endpoint.find('/', pos+2);
+            size_t pos_end = endpoint.find('/',pos_begin+1);
```

Comment:
These two lines are identical in both if-else path, so can be moved before the if.
And better to check that they are both not npos.

Reply:
I too thought of combining it but looks like we need different check, so kept them separate. Please let me know if you have any suggestions